### ISSUE TITLE
Refactor _get_steplength

### ISSUE THREAD

**Issue by tommyod** [equinor/iterative_ensemble_smoother#107]
Instead of having 3 arguments go into `SIES` that determine the step length from an exponential decay given by `_get_steplength`, I propose that the user supplies a callback. This is the approach taken in e.g. [keras](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/LearningRateScheduler). The advantage is:

- More general approach, if a user wants to try a different schedule, it is more easily done.
- Hides implementation details from the user, the 3 variables max_steplength, min_steplength, dec_steplength are not really important for most users. Nor do they pertain to the SIES itself, they are only related to one particular part of it.

In pseudocode, I propose:

```python
def exponential_decay(iteration, learning_rate)
    # a schedule function takes an iteration index (integer, indexed from 0) and 
    # current learning rate (float) as inputs and returns a new learning rate as output (float).
    # logic here
    return learning_rate


class SIES:

    def __init__(
        self,
        ensemble_size: int,
        *,
        schedule: Union[callable, None],
    ):
    if schedule is None:
        self.schedule = exponential_decay
    else:
        self.schedule  = schedule 

```
